### DISCUSSION TITLE
About the divergence in the way max collisions works between Godot Physics and Godot Jolt.

### DISCUSSION THREAD

**Discussion by yosoyfreeman** [godot-jolt/godot-jolt#804] [General]
Hi there, I hope you are having a great day!

I'm looking for any breaking changes for the time Godot jolt becomes official and working on my own move and slide.

I noticed that there is a different in the way collisions are reported. With Godot jolt, you get a collision for each plane you depenetrated against (Except with the issues of missing collisions right now), but with Godot physics, you get a collision per collision point, in the same way you would have with native physics collisions.

So in practical terms this means that in Godot jolt, a downwards velocity against a floor will report one collision, but in Godot it varies depending of the shape: A capsule only produces one contact point, a box 4, and a cylinder 12. 

Is this expected behaviour? I find the way Jolt report kinematic collisions makes more sense, but the divergence is there and i didn't see it pointed anywhere. It may be something worth documenting, because is one of the few divergences i found that can drastically alter the gameplay and will need changes going from Godot physics to Godot Jolt.

**Comment by mihe** [comment]
I've run into this discrepancy once or twice when debugging `body_test_motion`/`test_move`, but I mostly just chalked it up to the fact that Godot and Jolt uses different types of collision detection, where Godot uses SAT (with GJK/EPA as fallback sometimes) and Jolt uses GJK/EPA exclusively.

But as you point out, this is in fact not the behavior you see with the normal contact manifolds, even when using Jolt, where you'll get ~4 contact points when stacking two boxes on top of eachother.

Now I'm curious as to why that is exactly. It might very well be that I messed something up with the collision collectors.

**Reply by mihe** [reply]
Hm, I'm not seeing any obvious way of getting a full manifold from the narrow-phase queries. Even if I gather all the contacts from a query of two overlapping boxes I only end up with a single hit/contact.

@jrouwe Does this make sense to begin with? I see that there's `ManifoldBetweenTwoFaces` and `PruneContactPoints` to form the manifold once you have the contact points, but is it possible to actually get the full set of points from just a single query alone? How is this achieved during normal simulation?

**Reply by mihe** [reply]
Nevermind, I'm an idiot. I completely misread the interface of `ManifoldBetweenTwoFaces` as taking two sets of `ContactPoints` for some reason.

I see now that it uses the supporting faces to create a manifold. I'll tinker around with that and see if I can achieve something that actually outputs the full manifold.

**Reply by yosoyfreeman** [reply]
Thank you for looking into it! If i can help with anything hit me!

**Comment by mihe** [comment]
Fixed in #806.
